SwitchSpeak-iOS/SwitchSpeak-iOS/TouchGrid/Tree/Node.swift

```diff
@@ -47,7 +47,31 @@ class Node {
 		for childNode in self.childNodes {
 			childNode.unHighlightSubTree()
 		}
-	}	
+	}
+    
+    // it will retrun a node. if it has only nonDummy child node, it will return that child node, else it will return itself
+    func mergeNode() -> Node {
+        var nonDummyChildCount = 0
+        var nonDummyChildren : Node = Node()
+        for childNode in self.childNodes {
+            if childNode.dummy == false {
+                nonDummyChildCount = nonDummyChildCount + 1
+                nonDummyChildren = childNode
+            }
+        }
+        if nonDummyChildCount == 1 {
+            return nonDummyChildren
+        }
+        else{
+            return self
+        }
+    }
+    // it will make every child node shrink if its has only single nonDummy child node
+    func ShrinkChildren() {
```

Comment:
Please change this to camelCase: `shrinkChildren`. Only classes, structs, and enums should be in TitleCase: `Node`, `TouchGrid`, etc.